src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -360,16 +387,23 @@ protected void InitializeHttpClient(HttpClientHandler httpClientHandler, params
                     {
                         handler = handler.InnerHandler as DelegatingHandler;
                     }
+
                     handler.InnerHandler = currentHandler;
                     currentHandler = handlers[i];
                 }
             }
 
-            var newClient = new HttpClient(currentHandler, false);
+            if (httpClient == null)
+            {
+                HttpClient = new HttpClient(currentHandler, false);
+            }
+            else
+            {
+                HttpClient = httpClient;
+            }
+
             FirstMessageHandler = currentHandler;
-            HttpClient = newClient;
-            Type type = this.GetType();
-            SetUserAgent(type.FullName, ClientVersion);
+            SetUserAgent(this.GetType().FullName, ClientVersion);
```

Comment:
@markcowl my idea of bare minimum change that we discussed for this feature would be not to alter lot of things here.
My idea was that defaultheaders will be untouched, but request headers can be used to pass in custom information (if needed)